app/controllers/offers_controller.rb

```diff
@@ -23,6 +24,13 @@ def destroy
 
   def cancel
     @offer.canceled!
+    @suggested_lot = @offer.suggested_lot
+    @requested_lot = @offer.requested_lot
+    if @offer.suggested_lot.user == current_user
+      NotificationSendingService.new('sender-cancel-offer', @offer.suggested_lot, @offer.requested_lot).call
+    elsif @offer.requested_lot.user == current_user
+      NotificationSendingService.new('recipient-cancel-offer', @offer.requested_lot, @offer.suggested_lot).call
+    end
```

Comment:
```suggestion
    if @suggested_lot.user == current_user
      NotificationSendingService.new('sender-cancel-offer', @suggested_lot, @requested_lot).call
    elsif @requested_lot.user == current_user
      NotificationSendingService.new('recipient-cancel-offer', @requested_lot, @suggested_lot).call
    end
```

Reply:
Btw, there is no view for this action, so why have `@requested_lot` and `@suggested_lot` as object's variables? Remove `@`. And check other controllers and actions in this PR for the same thing.